backend/app/deps/authorization_deps.py

```diff
@@ -227,7 +254,15 @@ async def __call__(
                     detail=f"User `{current_user} does not have `{self.role}` permission on file {file_id}",
                 )
             else:
-                raise HTTPException(status_code=404, detail=f"File {file_id} not found")
+                public_access = await check_public_access(
+                    file_id, "file", self.role, current_user
+                )
+                if public_access:
+                    return True
```

Comment:
I did it like this since it looked like the method was supposed to either return True or throw an error. But I did change it to returning public_access instead of 'True' since that looked confusing.